### PR TITLE
[CAS-578] Add options after clicking user on GroupChatInfoFragment

### PR DESCRIPTION
**Pull request by adasiewiczr**
| Dark | Light |
| --- | --- |
|Screenshot 2021-01-29 at 09 16 32|Screenshot 2021-01-29 at 09 17 01|

https://user-images.githubusercontent.com/17440581/106251445-449e6a80-6215-11eb-917f-722a87bef2ce.mp4

Adds menu after clicking user on GroupChatInfoFragment:
1. View Info - opens ChatInfo screen with hidden options when one to one conversation doesn't exist
2. Message - opens Chat screen if exists and ChatPreview screen which creates a draft channel and converts it to channel after sending the first message

What's not covered in this PR:
- Header's title on ChatInfo screen is not centered
- Icons on ChatInfo screen have wrong colors

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/preview/ChatPreviewViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private var cid: String? = null
+    private val _state: MutableLiveData<State> = MutableLiveData()
+    private val _events: MutableLiveData<Event<UiEvent>> = MutableLiveData()
+    val state: LiveData<State> = _state
+    val events = _events
+
+    init {
+        _state.value = State(cid = null)
+        viewModelScope.launch {
+            val result = ChatClient.instance()
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/preview/ChatPreviewViewModel.kt:30]: Why do you use ChatClient directly? Why not `UseCaseHelper::createChannel`?

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                filter = Filters.and(
+                    Filters.eq("type", "messaging"),
+                    FilterObject(
+                        data = mutableMapOf(
+                            "distinct" to true,
+                            "members" to listOf(memberId, chatDomain.currentUser.id),
+                        ),
+                    ),
+                ),
+                sort = QuerySort.desc("last_updated"),
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt:47]: Better to use such syntax `QuerySort.desc(Channel::lastUpdated)`

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                messageLimit = 0,
+                limit = 1,
+            ).await()
+            if (result.isSuccess) {
+                _state.addSource(
+                    Transformations.map(result.data().channelsState) { channelState ->
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt:53]: What if for better redabillity move inner lambda to some method? E.g. `fun mapChannelState(channelState: ChannelState): State`?

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                UiEvent.RedirectToChat(state.directChannelCid)
+            } else {
+                UiEvent.RedirectToChatPreview
+            },
+        )
+    }
+
+    private fun removeFromChannel() {
+        viewModelScope.launch {
+            val result = channelClient.removeMembers(memberId).await()
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt:97]: We need to add offline SDK support for this use case. If it's not a part of this PR, then it's good to create a new task and put it in our backlog

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/preview/ChatPreviewViewModel.kt (hunk 1 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/preview/ChatPreviewViewModel.kt:30]: `CreateChannel` use case doesn't support creating `distinct` channels (based on members and not on id and type). I'll task for that

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt (hunk 4 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt:97]: https://stream-io.atlassian.net/browse/CAS-648

**Review by ogkuzmin — APPROVED** [review]

**Pull request merged**